akka-docs-dev/rst/scala/http/common/http-model.rst

```diff
@@ -0,0 +1,221 @@
+warning message is logged depending on the value of the ``illegal-header-warnings`` config setting.
+
+Some headers have special status in HTTP and are therefore treated differently from "regular" headers:
+
+Content-Type
+  The Content-Type of an HTTP message is modeled as the ``contentType`` field of the ``HttpEntity``.
+  The ``Content-Type`` header therefore doesn't appear in the ``headers`` sequence of a message.
+  Also, a ``Content-Type`` header instance that is explicitly added to the ``headers`` of a request or response will
+  not be rendered onto the wire and trigger a warning being logged instead!
+
+Transfer-Encoding
+  Messages with ``Transfer-Encoding: chunked`` are represented via the ``HttpEntity.Chunked`` entity.
+  As such chunked messages that do not have another deeper nested transfer encoding will not have a ``Transfer-Encoding``
+  header is their ``headers`` sequence.
+  Similarly, a ``Transfer-Encoding`` header instance that is explicitly added to the ``headers`` of a request or
+  response will not be rendered onto the wire and trigger a warning being logged instead!
+
+Content-Length
+  The content length of a message is modelled via its :ref:`HttpEntity`. As such no ``Content-Length`` header will ever
+  be part of a message's ``header`` sequence.
+  Similarly, a ``Content-Length`` header instance that is explicitly added to the ``headers`` of a request or
+  response will not be rendered onto the wire and trigger a warning being logged instead!
+
+Server
+  A ``Server`` header is usually added automatically to any response and its value can be configured via the
+  ``akka.http.server.server-header`` setting. Additionally an application can override the configured header with a
+  custom one by adding it to the response's ``header`` sequence.
+
+User-Agent
+  A ``User-Agent`` header is usually added automatically to any request and its value can be configured via the
+  ``akka.http.client.user-agent-header`` setting. Additionally an application can override the configured header with a
+  custom one by adding it to the request's ``header`` sequence.
+
+Date
+  The ``Date`` response header is added automatically but can be overridden by supplying it manually.
+
+Connection
+  On the server-side Akka HTTP watches for explicitly added ``Connection: close`` response headers and as such honors
+  the potential wish of the application to close the connection after the respective response has been sent out.
+  The actual logic for determining whether to close the connection is quite involved. It takes into account the
+  request's method, protocol and potential ``Connection`` header as well as the response's protocol, entity and
+  potential ``Connection`` header. See `this test`__ for a full table of what happens when.
+
+__ @github@/akka-http-core/src/test/scala/akka/http/impl/engine/rendering/ResponseRendererSpec.scala#L422
```

Comment:
Actually I think that documentation should link to the code a lot more than is usually done.
If only links wouldn't have the strong tendency to become stale...